src/uart.toit

```diff
@@ -0,0 +1,277 @@
+// Copyright (C) 2024 Toitware ApS. All rights reserved.
+// Use of this source code is governed by an MIT-style license that can be
+// found in the LICENSE file.
+
+import binary show LITTLE-ENDIAN
+import encoding.base64
+import encoding.json
+import encoding.ubjson
+import gpio
+import log
+import uart
+import reader show BufferedReader Reader
+import system
+
+import .jaguar
+
+class EndpointUart implements Endpoint:
+  config_/Map
+  logger/log.Logger
+
+  constructor --config/Map --logger/log.Logger:
+    config_ = config
+    this.logger = logger.with-name "uart"
+
+  run device/Device:
+    print "running uart endpoint"
+    logger.debug "running uart endpoint"
+    rx := gpio.Pin config_["rx"]
+    tx := gpio.Pin config_["tx"]
+    port := uart.Port
+        --rx=rx
+        --tx=tx
+        --baud-rate=(config_.get "baud-rate") or 115200
```

Comment:
Not entirely sure about this. It isn't exactly the same and maybe having a null entry should be the same as not having one.

```suggestion
        --baud-rate=config_.get "baud-rate" --if-absent=: 115200
```